examples/resnet34_imagenet.py

```diff
@@ -0,0 +1,295 @@
+                               kernel_initializer=init_residual)(input_layer)
+    x = tf.keras.layers.ReLU()(x)
+    x = tf.keras.layers.Dropout(0.25)(x)
+    x = tf.keras.layers.Conv2D(filters, 3, padding='same', strides=1, use_bias=False,
+                               kernel_initializer=init_residual)(x)
+
+    if downsample:
+        res = tf.keras.layers.Conv2D(filters, 1, padding='same', strides=2, use_bias=False)(input_layer)
+    else:
+        res = input_layer
+
+    x = tf.keras.layers.Add()([x, res])
+    x = tf.keras.layers.ReLU()(x)
+
+    return x
+
+
+def resnet34():
+    inputs = tf.keras.layers.Input(shape=(224, 224, 3))
+
+    x = tf.keras.layers.Conv2D(64, 3, padding='same', strides=1, use_bias=False,
+                               kernel_initializer=init_non_residual)(inputs)
+    x = tf.keras.layers.ReLU()(x)
+
+    # Sengupta et al (2019) use two extra plain pre-processing layers,
+    # so this technically isn't ResNet34.
+    x = tf.keras.layers.Conv2D(64, 3, padding='same', strides=1, use_bias=False,
+                               kernel_initializer=init_non_residual)(x)
+    x = tf.keras.layers.ReLU()(x)
+    x = tf.keras.layers.Conv2D(64, 3, padding='same', strides=2, use_bias=False,
+                               kernel_initializer=init_non_residual)(x)
+    x = tf.keras.layers.ReLU()(x)
+    x = tf.keras.layers.AveragePooling2D(2)(x)
+
+    x = resnet34_block(x, 64)
+    x = resnet34_block(x, 64)
+    x = resnet34_block(x, 64)
+
+    x = resnet34_block(x, 128, True)
+    x = resnet34_block(x, 128)
+    x = resnet34_block(x, 128)
+    x = resnet34_block(x, 128)
+
+    x = resnet34_block(x, 256, True)
+    x = resnet34_block(x, 256)
+    x = resnet34_block(x, 256)
+    x = resnet34_block(x, 256)
+    x = resnet34_block(x, 256)
+    x = resnet34_block(x, 256)
+
+    x = resnet34_block(x, 512, True)
+    x = resnet34_block(x, 512)
+    x = resnet34_block(x, 512)
+
+    x = tf.keras.layers.GlobalAveragePooling2D()(x)
+    x = tf.keras.layers.Dense(1000, use_bias=False)(x)
+    x = tf.keras.layers.Softmax()(x)
+
+    model = tf.keras.Model(inputs=inputs, outputs=x, name='resnet34_imagenet')
+
+    return model
+
+
+def parse_buffer(buffer):
+    keys_to_features = {
+        "image/encoded": tf.io.FixedLenFeature((), tf.string, ''),
+        "image/class/label": tf.io.FixedLenFeature([], tf.int64, -1)}
+    parsed = tf.io.parse_single_example(buffer, keys_to_features)
+
+    # get label
+    label = tf.cast(tf.reshape(parsed["image/class/label"], shape=[]), dtype=tf.int32) - 1
+
+    # decode image
+    image = tf.image.decode_jpeg(tf.reshape(parsed["image/encoded"], shape=[]), channels=3)
+
+    # convert to float **NOTE** this converts [0,255] to [0,1) range
+    image = tf.image.convert_image_dtype(image, tf.float32)
+
+    return image, label
+
+
+#if __name__ == '__main__':
+with tf.device('/CPU:0'):
+    for gpu in tf.config.experimental.list_physical_devices('GPU'):
```

Comment:
I also have a vague memory that this will crash if you've already forced the device